specs/ExecuteScript.md

```diff
@@ -0,0 +1,222 @@
+
+  /// Remove an event handler previously added with `add_NavigationStarting`.
+  HRESULT remove_NavigationStarting(
+      [in] EventRegistrationToken token);
+
+  /// Add an event handler for the `ContentLoading` event.  `ContentLoading`
+  /// triggers before any content is loaded, including scripts added with
+  /// `AddScriptToExecuteOnDocumentCreated`.  `ContentLoading` does not trigger
+  /// if a same page navigation occurs (such as through `fragment`
+  /// navigations or `history.pushState` navigations).  This operation
+  /// follows the `NavigationStarting` and precedes `NavigationCompleted` events.
+  HRESULT add_ContentLoading(
+      [in] ICoreWebView2FrameContentLoadingEventHandler* eventHandler,
+      [out] EventRegistrationToken* token);
+
+  /// Remove an event handler previously added with `add_ContentLoading`.
+  HRESULT remove_ContentLoading(
+      [in] EventRegistrationToken token);
+
+  /// Add an event handler for the `NavigationCompleted` event.
+  /// `NavigationCompleted` runs when the CoreWebView2Frame has completely
+  /// loaded (`body.onload` runs) or loading stopped with error.
+  ///
+  /// \snippet ScenarioWebViewEventMonitor.cpp NavigationCompleted
+  HRESULT add_NavigationCompleted(
+      [in] ICoreWebView2FrameNavigationCompletedEventHandler* eventHandler,
+      [out] EventRegistrationToken* token);
+
+  /// Remove an event handler previously added with `add_NavigationCompleted`.
+  HRESULT remove_NavigationCompleted(
+      [in] EventRegistrationToken token);
+
+  /// Add an event handler for the DOMContentLoaded event.
+  /// DOMContentLoaded is raised when the iframe html document has been parsed.
+  /// This aligns with the the document's DOMContentLoaded event in html.
+  ///
+  /// \snippet ScenarioWebViewEventMonitor.cpp DOMContentLoaded
+  HRESULT add_DOMContentLoaded(
+      [in] ICoreWebView2FrameDOMContentLoadedEventHandler* eventHandler,
+      [out] EventRegistrationToken* token);
+  /// Remove an event handler previously added with add_DOMContentLoaded.
+  HRESULT remove_DOMContentLoaded(
+      [in] EventRegistrationToken token);
+
+  /// Run JavaScript code from the javascript parameter in the current frame.
+  /// The result of evaluating the provided JavaScript is used in this parameter.
+  /// The result value is a JSON encoded string. If the result is undefined,
+  /// contains a reference cycle, or otherwise is not able to be encoded into
+  /// JSON, the JSON `null` value is returned as the `null` string.
+  ///
+  /// \> [!NOTE]\n\> A function that has no explicit return value returns undefined.  If the
+  /// script that was run throws an unhandled exception, then the result is
+  /// also `null`. This method is applied asynchronously. If the method is
+  /// run before `ContentLoading`, the script will not be executed
+  /// and the JSON `null` will be returned.
+  /// This operation works even if `ICoreWebView2Settings::IsScriptEnabled` is
```

Comment:
By "works", do you mean "The operation doesn't crash -- returns `S_OK` (but doesn't do anything and immediately completes with `"null"`)", or do you mean "The operation executes the script"?

Reply:
It executes script. Please make clearer (and also please update ICoreWebView2::Executescript)